src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchForCompletableConsumerTest.java

```diff
@@ -0,0 +1,115 @@
+import io.restassured.http.ContentType;
+import net.serenitybdd.rest.SerenityRest;
+import org.junit.jupiter.api.Test;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import uk.gov.hmcts.reform.wataskmanagementapi.SpringBootContractBaseTest;
+import uk.gov.hmcts.reform.wataskmanagementapi.provider.service.CamundaConsumerApplication;
+import uk.gov.hmcts.reform.wataskmanagementapi.provider.service.TaskManagementProviderTestConfiguration;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.pactfoundation.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@PactTestFor(providerName = "wa_task_management_api_search_completable", port = "8991")
+@ContextConfiguration(classes = {CamundaConsumerApplication.class})
+@Import(TaskManagementProviderTestConfiguration.class)
+public class TaskManagementGetTasksBySearchForCompletableConsumerTest extends SpringBootContractBaseTest {
+
+    public static final String CONTENT_TYPE = "Content-Type";
+    private static final String WA_URL = "/task";
+    private static final String WA_SEARCH_FOR_COMPLETABLE = WA_URL + "/search-for-completable";
+
+    @Test
+    @PactTestFor(pactMethod = "executeSearchForCompletable200")
+    void testSearchForCompletable200Test(MockServer mockServer) throws IOException {
+        SerenityRest
+            .given()
+            .headers(getHttpHeaders())
+            .contentType(ContentType.JSON)
+            .body(creteSearchEventCaseRequest())
+            .post(mockServer.getUrl() + WA_SEARCH_FOR_COMPLETABLE)
+            .then()
+            .statusCode(HttpStatus.OK.value());
+    }
+
+    @Pact(provider = "wa_task_management_api_search_completable", consumer = "wa_task_management_api")
+    public RequestResponsePact executeSearchForCompletable200(PactDslWithProvider builder) {
+        return builder
+            .given("appropriate tasks are returned by search for completable")
+            .uponReceiving("Provider receives a POST /task/search-for-completable request from a WA API")
+            .path(WA_SEARCH_FOR_COMPLETABLE)
+            .method(HttpMethod.POST.toString())
+            .headers(getTaskManagementServiceResponseHeaders())
+            .matchHeader(AUTHORIZATION, AUTH_TOKEN)
+            .matchHeader(SERVICE_AUTHORIZATION, SERVICE_AUTH_TOKEN)
+            .body(creteSearchEventCaseRequest(), String.valueOf(ContentType.JSON))
+            .willRespondWith()
+            .status(HttpStatus.OK.value())
+            .body(createResponseForGetTask())
+            .toPact();
+    }
+
+    private DslPart createResponseForGetTask() {
+        return newJsonBody(
+            o -> o
+            .booleanValue("task_required_for_event", false)
```

Comment:
can we use booleanType here so it passes regardless of true of false being passed by consuming service.